CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+cmake_minimum_required (VERSION 3.23)
+
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake"
```

Comment:
CMake min required needs to be set to *something*. We can reassess after the other fixes. CMAKE_TOOLCHAIN_FILE depends on how you're going to work on it, but it's probably correct to remove it from the repo.

Reply:
I just double-checked and I set min-required to 3.23 for a reason, it seems. The `FILE_SET headers` directive in the `install(TARGETS)` command wasn't introduced until then.

Reply:
Yes, indeed, I noticed the same that 3.23 was necessary because of the `FILE_SET`. But should we perhaps reconsider whether this is absolutely necessary? It depends a bit on what systems we would like to (easily) support. Many systems do not have CMake 3.23 yet (see https://repology.org/project/cmake/versions), so this would require them to install it manually.

Reply:
This is up to you, really. My take is that because cmake is both only required for building from source and is also extremely portable w/r/t ABI compatibility I personally prefer using the newer tooling if at all possible.  The correct version to target if you decide it is valuable is probably 3.20 as it appears to be the version available for RHEL 8 and the other common long term support distros vendor the same or later versions. It shouldn't be too difficult to change from FILE_SET now that I merged the include and src directories.